qiskit_optimization/applications/bin_packing.py

```diff
@@ -31,14 +31,19 @@ class BinPacking(OptimizationApplication):
         https://en.wikipedia.org/wiki/Bin_packing_problem
     """
 
-    def __init__(self, weights: List[int], max_weight: int) -> None:
+    def __init__(
+        self, weights: List[int], max_weight: int, max_number_of_bins: Optional[int] = None
+    ) -> None:
         """
         Args:
             weights: A list of the weights of items
             max_weight: The maximum bin weight capacity
+            max_number_of_bins: The maximum number of bins
         """
         self._weights = weights
         self._max_weight = max_weight
+        if max_number_of_bins is None:
+            self.max_number_of_bins = len(weights)
```

Comment:
Should this be self._max_number_of_bins so its marked as private variable like the others (I don't think we want to expose this as part of the public API right - which as it stands, not being marked private, it appears to be)